### PR TITLE
fix: RedirectUri test-coverage

### PR DESCRIPTION
**Pull request by jmunozherbas**
NOTE: This PR replaces https://github.com/JanssenProject/jans/pull/767 

1.- Fixed, all sonar issues
2.- inproved, code coverage percent with unit test ( methods with logic)

### CODE REVIEW

**Review by yuriyz — DISMISSED** [review]

**Comment by yuriyz** [comment]
@jmunozherbas changes looks good but merge is blocked because your commit is not signed. Please sign your commits.
https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

File: jans-auth-server/pom.xml [hunk 1]
```diff
@@ -366,6 +366,18 @@
...
+				<artifactId>mockito-core</artifactId>
+				<version>3.11.2</version>
+				<scope>test</scope>
+			</dependency>
+			<dependency>
+				<groupId>org.mockito</groupId>
+				<artifactId>mockito-testng</artifactId>
+				<version>0.4.8</version>
+				<scope>test</scope>
+			</dependency>
```
Comment (Milton-Ch) [inline jans-auth-server/pom.xml:380]: Looks like these ones can go in `jans-bom/pom.xml` and here you can import them without version, same thing than testng dependency.

File: jans-auth-server/common/src/test/java/io/jans/as/common/util/RedirectUriTest.java [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+        assertEquals(redirectUri2.getResponseParamentersSize(), paramsSize);
+    }
+
+    private String encodeUTF8(String input) throws UnsupportedEncodingException {
+        return URLEncoder.encode(input, Util.UTF8_STRING_ENCODING);
+    }
+
+    private RSAPublicKey getRSAPublicKey() throws NoSuchAlgorithmException, InvalidKeySpecException {
+        String big1 = "00e0b509f6259df8642dbc35662901477df22677ec152b5ff68ace615bb7b725152b3ab17a876aea8a5aa76d2e417629ec4ee341f56135fccf695280104e0312ecbda92557c93870114af6c9d05c4f7f0c3685b7a46bee255932575cce10b424d813cfe4875d3e82047b97ddef52741d546b8e289dc6935b3ece0462db0a22b8e7";
+        String big2 = "010001";
```
Comment (Milton-Ch) [inline jans-auth-server/common/src/test/java/io/jans/as/common/util/RedirectUriTest.java]: Minor: just clearer names, for instace here: `modulus` and `publicExponent`, just to make them easy to guess, however it's a minor thing.

File: jans-auth-server/common/src/test/java/io/jans/as/common/util/RedirectUriTest.java [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+        assertNoEmptyQueryString(queryResult, RESPONSE, 1);
+        assertTrue(queryResult.startsWith("response=eyJ0eXAiOiJqd3QiLCJlbmMiOiJBMTI4R0NNIiwiYWxnIjoiQTEyOEtXIn0."));
+    }
+
+    @Test
+    public void getQueryString_withEncriptionAlgorithmRSAAndSignatureAlgorithm_responseEncoded() throws UnsupportedEncodingException, CryptoProviderException, NoSuchAlgorithmException, InvalidKeySpecException {
+        RedirectUri redirectUri = getRedirectUriTemplateGetQueryString(ResponseMode.JWT, KeyEncryptionAlgorithm.RSA1_5, BlockEncryptionAlgorithm.A256GCM, SignatureAlgorithm.HS256);
+        AbstractCryptoProvider cryptoProvider = mock(AbstractCryptoProvider.class);
+        when(cryptoProvider.getPublicKey(anyString(), Mockito.any(), Mockito.any())).thenReturn(getRSAPublicKey());
+        when(cryptoProvider.sign(anyString(), anyString(), anyString(), any())).thenReturn("12345");
```
Comment (Milton-Ch) [inline jans-auth-server/common/src/test/java/io/jans/as/common/util/RedirectUriTest.java]: Usually when you add `when`s, you should add `verify`s at the end as well and it's good to add `verifyNoMoreInteractions` finally for those mocked classes. That way you ensure that the flow was as expected. Worths checking if you can validate one or two params.

File: jans-auth-server/common/src/test/java/io/jans/as/common/util/RedirectUriTest.java [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+        assertTrue(queryResult.length() > 0);
+        assertEquals(queryResult, "eyJraWQiOiJrZXkxMjMiLCJ0eXAiOiJqd3QiLCJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJ1c2VyMTIzIiwiZXhwIjoiMzAwMCIsImV4cGlyZXNfaW4iOiIzMDAwIn0.12345");
+    }
+
+    @Test
+    public void getQueryString_withEncriptionAlgorithm128PublicKeyInvalid_responseEmptyThrowInvalid() throws CryptoProviderException {
+        RedirectUri redirectUri = getRedirectUriTemplateGetQueryString(ResponseMode.JWT, KeyEncryptionAlgorithm.RSA1_5, BlockEncryptionAlgorithm.A256GCM, null);
+
+        AbstractCryptoProvider cryptoProvider = mock(AbstractCryptoProvider.class);
+        when(cryptoProvider.getPublicKey(anyString(), Mockito.any(), Mockito.any())).thenReturn(null);
```
Comment (Milton-Ch) [inline jans-auth-server/common/src/test/java/io/jans/as/common/util/RedirectUriTest.java]: Minor: you can remove `Mockito.`, it's already imported.

**Review by Milton-Ch — DISMISSED** [review]
Overall looks good, just added minor things, feel free to update or not based on comments, thanks.

**Pull request merged**